script/model/forecast.py

```diff
@@ -0,0 +1,520 @@
+        will be the number of data points fall into an interval for each interval in the original dataset
+
+        :param data: Time-series data with TIMESTAMP column
+        :param interval_ms: Interval duratoin in MS
+        :return:
+        """
+        start_t = data[0][self.TS_IDX]
+        end_t = data[-1][self.TS_IDX]
+
+        if end_t - start_t <= 1:
+            return None
+
+        # Number of data points in the new time-series
+        num_buckets = (end_t - start_t - 1) // interval_ms + 1
+
+        new_data = np.zeros(num_buckets)
+
+        for i in range(len(data)):
+            t = data[i][self.TS_IDX]
+            bi = (t - start_t) // interval_ms
+
+            new_data[bi] += 1
+
+        new_data = new_data[LOAD_PHASE_NUM_DATA:]
+        return new_data
+
+    def _split_data(self, data: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Split the raw data into a training set, and a testing(evaluation) set.
+        :param data: All the raw data
+        :return: traing, test raw data set
+        """
+        if self.test_mode:
+            self.test_set_size = len(data)
+        else:
+            self.test_set_size = self.eval_data_size
+
+        split_idx = len(data) - self.test_set_size
+        # First part as the training set
+        train_raw_data = data[:split_idx]
+
+        # Last part as the testing set
+        test_raw_data = data[split_idx:]
+
+        return train_raw_data, test_raw_data
+
+    def create_seqs(self) -> List[Tuple[torch.Tensor, torch.Tensor]]:
+        """
+        Create training sequences.  Each training sample consists of:
+         - A list of data points as input
+         - A label (target) value in some horizon
+        :return: training sequences
+        """
+        seq_len = self.seq_len
+        horizon = self.horizon_len
+        train_norm_data = self._normalize_data(self.train_raw_data)
+
+        input_data = torch.FloatTensor(train_norm_data).view(-1)
+        seqs = []
+        for i in range(len(input_data) - seq_len - horizon):
+            seq = input_data[i:i + seq_len]
+
+            # Look beyond the horizon to get the label
+            label_i = i + seq_len + horizon
+            label = input_data[label_i: label_i + 1]
+
+            seqs.append((seq, label))
+
+        return seqs
+
+    def get_test_input(self) -> List[float]:
+        """
+        Get the test input (normalized)
+        :return: test time-series data
+        """
+        test_input_norm = self._normalize_data(self.test_raw_data)
+        test_input_list = test_input_norm.tolist()
+        return test_input_list
+
+    def stat_util(self) -> None:
+        trace_start_ms = self.data[0][1]
+        trace_end_ms = self.data[-1][1]
+        trace_nsample = len(self.data)
+
+        print(
+            f"{trace_nsample} samples from {trace_start_ms} to {trace_end_ms}\n"
+            f"{(trace_end_ms - trace_start_ms) / 1000000}seconds")
+
+
+class LSTM(nn.Module):
```

Comment:
Create another `Model` wrapper class similar to `script/model/model.py` with `LSTM` only as one implementation option.  `LSTM` should just provide the simple sklearn style `fit(x, y)` and `predict(x)` interface. This way it's very easy to extend to other models.